src/core/public/http/types.ts

```diff
@@ -17,10 +17,30 @@
  * under the License.
  */
 
-import { BasePathSetup } from '../base_path';
+import { Observable } from 'rxjs';
 import { InjectedMetadataSetup } from '../injected_metadata';
 import { FatalErrorsSetup } from '../fatal_errors';
 
+export interface HttpServiceBase {
+  stop(): void;
+  getBasePath(): string;
+  addToPath(path: string): string;
+  removeFromPath(path: string): string;
+  fetch: HttpHandler;
+  delete: HttpHandler;
+  get: HttpHandler;
+  head: HttpHandler;
+  options: HttpHandler;
+  patch: HttpHandler;
+  post: HttpHandler;
+  put: HttpHandler;
+  addLoadingCount(count$: Observable<number>): void;
+  getLoadingCount$(): Observable<number>;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface HttpSetup extends HttpServiceBase {}
```

Comment:
can be declared as `export type HttpSetup = HttpServiceBase;` then we don't need to mute empty interface error

Reply:
`HttpSetup` should be declared as `public` to be included in generated docs